pkg/edgedictionary/edgedictionary_test.go

```diff
@@ -0,0 +1,279 @@
+		ServiceID: i.Service,
+		Version:   i.Version,
+		Name:      i.Name,
+		CreatedAt: testutil.MustParseTimeRFC3339("2001-02-03T04:05:06Z"),
+		WriteOnly: false,
+		ID:        "456",
+		UpdatedAt: testutil.MustParseTimeRFC3339("2001-02-03T04:05:07Z"),
+		DeletedAt: testutil.MustParseTimeRFC3339("2001-02-03T04:05:08Z"),
+	}, nil
+}
+
+func createDictionaryOK(i *fastly.CreateDictionaryInput) (*fastly.Dictionary, error) {
+	return &fastly.Dictionary{
+		ServiceID: i.Service,
+		Version:   i.Version,
+		Name:      i.Name,
+		CreatedAt: testutil.MustParseTimeRFC3339("2001-02-03T04:05:06Z"),
+		WriteOnly: false,
+		ID:        "456",
+		UpdatedAt: testutil.MustParseTimeRFC3339("2001-02-03T04:05:07Z"),
+	}, nil
+}
+
+// These two are responses from other packages that will need to update if
+// the responses ever change.
+func getDictionaryInfoOK(i *fastly.GetDictionaryInfoInput) (*fastly.DictionaryInfo, error) {
+	return &fastly.DictionaryInfo{
+		ItemCount:   2,
+		LastUpdated: testutil.MustParseTimeRFC3339("2001-02-03T04:05:07Z"),
+		Digest:      "digest_hash",
+	}, nil
+}
+
+func listDictionaryItemsOK(i *fastly.ListDictionaryItemsInput) ([]*fastly.DictionaryItem, error) {
+	return []*fastly.DictionaryItem{
+		&fastly.DictionaryItem{
+			ServiceID:    i.Service,
+			DictionaryID: "456",
+			ItemKey:      "foo",
+			ItemValue:    "bar",
+			CreatedAt:    testutil.MustParseTimeRFC3339("2001-02-03T04:05:06Z"),
+			UpdatedAt:    testutil.MustParseTimeRFC3339("2001-02-03T04:05:07Z"),
+		},
+		&fastly.DictionaryItem{
+			ServiceID:    i.Service,
+			DictionaryID: "456",
+			ItemKey:      "baz",
+			ItemValue:    "bear",
+			CreatedAt:    testutil.MustParseTimeRFC3339("2001-02-03T04:05:06Z"),
+			UpdatedAt:    testutil.MustParseTimeRFC3339("2001-02-03T04:05:07Z"),
+			DeletedAt:    testutil.MustParseTimeRFC3339("2001-02-03T04:06:08Z"),
+		},
+	}, nil
+}
+
+func createDictionaryDuplicate(*fastly.CreateDictionaryInput) (*fastly.Dictionary, error) {
+	return nil, errors.New("Duplicate record")
+}
+
+func deleteDictionaryOK(*fastly.DeleteDictionaryInput) error {
+	return nil
+}
+
+func deleteDictionaryError(*fastly.DeleteDictionaryInput) error {
+	return errTest
+}
+
+var errTest = errors.New("an expected error ocurred")
+
+var createDictionaryOutput = "\nSUCCESS: Created dictionary denylist (service 123 version 1)\n"
+var deleteDictionaryOutput = "\nSUCCESS: Deleted dictionary allowlist (service 123 version 1)\n"
+
+var describeDictionaryOutput = strings.TrimSpace(`
+Service ID: 123
+Version: 1
+ID: 456
+Name: dict-1
+Write Only: false
+Created (UTC): 2001-02-03 04:05
+Last edited (UTC): 2001-02-03 04:05
+`) + "\n"
+
+var describeDictionaryOutputDeleted = strings.TrimSpace(`
+Service ID: 123
+Version: 1
+ID: 456
+Name: dict-1
+Write Only: false
+Created (UTC): 2001-02-03 04:05
+Last edited (UTC): 2001-02-03 04:05
+Deleted (UTC): 2001-02-03 04:05
+`) + "\n"
+
+var describeDictionaryOutputVerbose = strings.TrimSpace(`
+Fastly API token not provided
```

Comment:
Probably me just not understanding the current usage of the tool, but would we expect an API request to be successful without an API token provided? e.g. as far as I'm aware the relevant API endpoint still requires authentication.

Reply:
This is an existing pattern in the other test classes. I assume the verbose output is different because the api token is not present and is not needed for the purpose of the test.

Reply:
OK cool. Thanks for clarifying 👍🏻